tests/python_on_whales/components/test_compose.py

```diff
@@ -780,3 +781,56 @@ def test_compose_ls_project_multiple_statuses():
         assert sorted(project.config_files) == sorted(d.client_config.compose_files)
 
     d.compose.down(timeout=1)
+
+
+def test_docker_compose_up_remove_orphans():

```

Comment:
Thanks for the PR, you can only have orphans if you change the docker-compose.yml file and remove services in this file as far as I know. Maybe that's why you have issues here in the unit tests?

Reply:
Hi @gabrieldemarmiesse, that is a fast response, thank you!

This is exactly what I'm doing in that test, running docker compose up on a file with 2 services,
rewrite the file to include only one, re-run docker compose up, and the re-run docker compose up again using the new flag.

Currently I'm a bit confused why there are differences between my end and git env.
It looks like on my environment once I modify the file and run up again, docker.compose.ps() returns the container only for the remaining service, while for github actions env it returns the container for both services.

Here is what I'm using:
```
$ docker -v
Docker version 20.10.12, build 20.10.12-0ubuntu2~20.04.1
$ docker compose version
Docker Compose version v2.12.2
```
any idea/hint is welcome :)

I'll investigate more what happens actually, and get back with updates.

Reply:
It's quite possible that the version of compose you are using is not the same as ghe one in the CI. I think the once in the ci is much older

Reply:
Feel free to upgrade it y'if you need